src/main/java/se/sundsvall/messaging/integration/webmessagesender/WebMessageSenderIntegrationMapper.java

```diff
@@ -11,28 +11,30 @@
 @Component
 class WebMessageSenderIntegrationMapper {
 
-    CreateWebMessageRequest toCreateWebMessageRequest(final WebMessageDto dto) {
-        if (dto == null) {
-            return null;
-        }
-
-        return new CreateWebMessageRequest()
-            .partyId(dto.partyId())
-            .externalReferences(Optional.ofNullable(dto.externalReferences())
-                .map(externalReferences -> externalReferences.stream()
-                    .map(externalReference -> new ExternalReference()
-                        .key(externalReference.key())
-                        .value(externalReference.value()))
-                    .toList())
-                .orElse(null))
-            .message(dto.message())
-            .attachments(Optional.ofNullable(dto.attachments())
-                .map(attachments -> attachments.stream()
-                    .map(attachment -> new Attachment()
-                        .fileName(attachment.fileName())
-                        .mimeType(attachment.mimeType())
-                        .base64Data(attachment.base64Data()))
-                    .toList())
-                .orElse(null));
-    }
+	CreateWebMessageRequest toCreateWebMessageRequest(final WebMessageDto dto) {
+		return Optional.ofNullable(dto)
+			.map(webMessageDto ->
+				new CreateWebMessageRequest()
+					.partyId(dto.partyId())
+					.externalReferences(Optional.ofNullable(dto.externalReferences())
+						.map(externalReferences -> externalReferences.stream()
+							.map(externalReference -> new ExternalReference()
+								.key(externalReference.key())
+								.value(externalReference.value()))
+							.toList())
+						.orElse(null))
+					.message(dto.message())
+					.oepInstance(Optional.ofNullable(dto.oepInstance())
+						.map(CreateWebMessageRequest.OepInstanceEnum::fromValue)
+						.orElse(CreateWebMessageRequest.OepInstanceEnum.EXTERNAL))
+					.attachments(Optional.ofNullable(dto.attachments())
+						.map(attachments -> attachments.stream()
```

Comment:
Maybe extract the attachment mapping to a seperate method?

Reply:
I don't see much value in that, it will not be used anywhere else. Just adds another test.